flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -1086,8 +1086,8 @@ public void initializeState(FunctionInitializationContext context) throws Except
 		nextTransactionalIdHintState = context.getOperatorStateStore().getUnionListState(
 			NEXT_TRANSACTIONAL_ID_HINT_DESCRIPTOR_V2);
 
-		if (context.getOperatorStateStore().getRegisteredStateNames().contains(NEXT_TRANSACTIONAL_ID_HINT_DESCRIPTOR)) {
-			migrateNextTransactionalIdHindState(context);
+		if (context.getOperatorStateStore().getRegisteredStateNames().contains(NEXT_TRANSACTIONAL_ID_HINT_DESCRIPTOR.getName())) {
```

Comment:
@becketqin / @pnowojski With this change applied, the Kafka tests are failing. What's your take?

Reply:
Hard to tell. One would have to debug it to understand what's happening.

However maybe it's better to just clean up this code and remove the migration fallback path? This check is for people migrating from Flink versions <= 1.8, straight to 1.13. We could remove the whole field `NEXT_TRANSACTIONAL_ID_HINT_DESCRIPTOR` and `migrateNextTransactionalIdHintState` and just state in the release notes, that if you are using kafka producer you can not upgrade from Flink 1.8 to Flink 1.13 directly. You have to first upgrade to Flink `[1.9, 1.12]`.

Reply:
Can you even upgrade to 1.9/1.12 though given that this code is presumably broken and has been present in 1.9?

Reply:
Yes, there is a migration test. Either this code is working correctly, or the test is not covering for this issue. 

But since the test is failing after adding this `getName()`, maybe it was working correctly after all?